test/widgets/drive_detail_page_test.dart

```diff
@@ -141,8 +142,33 @@ void main() {
 
       final Finder profileWidget = find.byType(ProfileWidget);
       tester.tap(profileWidget.first);
+    });
+    testWidgets('can Navigate to chatPage on Waypoint', (WidgetTester tester) async {
+      drive.rides!.add(RideFactory().generateFake(
+        id: 1,
+        start: 'Waypoint',
+        status: RideStatus.approved,
+      ));
+      whenRequest(processor).thenReturnJson(drive.toJsonForApi());
+      whenRequest(processor,
+              urlMatcher: equals(equals(
+                  '/rest/v1/messages?select=%2A&chat_id=eq.${drive.rides!.last.chatId}&order=created_at.desc.nullslast')))
+          .thenReturnJson([]);
+      await pumpMaterial(tester, DriveDetailPage.fromDrive(drive));
 
-      // Add navigating to chat here
+      await tester.pump();
+      await pumpMaterial(tester, DriveDetailPage.fromDrive(drive));
+      await tester.pump();
+      await tester.tap(find.byKey(Key('chatPageButton${drive.rides!.last.id}Start')));
+      await tester.pumpAndSettle();
+      await tester.pump();
+      await tester.pumpAndSettle();
```

Comment:
this whole section is really weird. Way to many pumps for my taste lol. Probably forgot to clean up after testing around?

Reply:
yes thanks that was the problem with the Hero xD